### ISSUE TITLE
Navigation block: Immediately insert block when only one block type is allowed

### ISSUE THREAD

**Issue by mtias** [WordPress/gutenberg#16659]
A nice quality of life change would be to tweak the inserter mechanisms for `inner blocks` areas so that if there is only one allowed block type, we insert it instead of opening the inserter.

This would allow creating blocks like "buttons row" where the inserter automatically adds a new child block of the specified type, simplifying insertion process.

**Comment by marekhrabe** [comment]
Looking more into this…

**Comment by marekhrabe** [comment]
I made a very terrible "button row" block to test with (will publish later) and I've noticed it would be really nice if the appender (plus button) inserted the only allowed block immediately instead of opening the full inserter with a single option. I'll also look into that.

Appender in my terrible button row block:

<img width="304" alt="Screenshot 2019-07-22 at 16 45 47" src="https://user-images.githubusercontent.com/156676/61641995-72116c00-aca0-11e9-8558-62945b8a80b9.png">

It could say "Add BlockName" in the single-allowed-block-type mode.

This is the inserter with only one option available. Not very useful:

<img width="304" alt="Screenshot 2019-07-22 at 16 48 06" src="https://user-images.githubusercontent.com/156676/61642040-86556900-aca0-11e9-8ba7-f2a77464df1a.png">

**Comment by draganescu** [comment]
@marekhrabe I added a PR, still work in progress I think that should implement this idea: #16708

**Comment by marekhrabe** [comment]
@draganescu Thanks for the PR! I wish you said something sooner. I've assigned the issue to myself, wrote a comment that I'm working on it and even shared some of my progress and yet we ended up working on the same thing in parallel without my knowledge, wasting half of my day :( 

Please say something next time, I'd be more than happy to pass the issue or collaborate on it.

**Comment by mtias** [comment]
@marekhrabe perhaps you can continue with the "Button Row" (or updating core Button) to use this? I'd love to have this: https://github.com/WordPress/gutenberg/issues/16480

**Comment by draganescu** [comment]
@marekhrabe yes this happens sometimes on GH, I am sorry!

I started as an exploration (the branch name is try/*) and shared it as work in progress. I mentioned you exactly because I noticed your comment :) but alas after pushing my code, and figured we could collaborate. Please, feel free to review #16708 and/or propose a better way as I am not sure my take is the best

**Comment by draganescu** [comment]
@shaunandrews this would also be another great place to discuss the opportunity of implementing this option. As we discussed [in this thread](https://github.com/WordPress/gutenberg/issues/17539#issuecomment-544860373) you made some very good points on why we shouldn't.

As @mtias is the author of this issue and I've also went ahead and implemented an overall update that allows for this feature to happen, independently of the Navigation block, we could arrive here at a conclusion on the steps for moving forward with this small feature.
